src/status_im/wallet/core.cljs

```diff
@@ -586,20 +587,40 @@
                 "watching-txs?" watching-txs?
                 "syncing-allowed?" syncing-allowed?
                 "watch-new-blocks?" watch-new-blocks?)
-      (if (and syncing-allowed?
+      (if (and (or syncing-allowed?
+                   ignore-syncing-settings?)
                (or
                 waiting?
                 force-start?
                 watching-txs?))
         (start-wallet cofx (boolean (or watch-new-blocks? watching-txs?)))
         (stop-wallet cofx)))))
 
+(fx/defn restart-wallet-service-default
+  [cofx]
+  (restart-wallet-service cofx nil))
+
 (fx/defn restart
   {:events [::restart]}
   [{:keys [db] :as cofx}]
-  (fx/merge
-   {:db (dissoc db :wallet-service/restart-timeout)}
-   (restart-wallet-service true false)))
+  (restart-wallet-service {:force-start? true}))
+
+(fx/defn restart-on-pull
+  {:events [:wallet.ui/pull-to-refresh-history]}
+  [{:keys [db now] :as cofx}]
+  (let [last-pull (get db :wallet/last-pull-time)
+        watching? (get db :wallet/watch-txs)]
+    (when (and (not watching?)
+               (or (not last-pull)
+                   (> (- now last-pull)
+                      (* 1 60 1000))))
```

Comment:
I would put this in a `(def ...)` with a comment, although is fairly clear what it is.

Reply:
done